test/spec/features/external_registry/list_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'spec_helper'
+
+describe 'external-registry ls' do
+  context 'when empty' do
+    it "outputs headers" do
+      k = run('kontena external registry ls')
+      expect(k.code).to be_zero
+      expect(k.out).to match(/NAME/)
+      expect(k.out.lines.size).to eq 1
+    end
+
+    context '-q' do
+      it "outputs nothing" do
+        k = run 'kontena external-registry ls -q'
+        expect(k.code).to be_zero
+        expect(k.out.strip).to be_empty
+      end
+    end
+  end
+
+  context 'when not empty' do
+    before do
+      run 'kontena external-registry add -u foo -e foo@no.email -p secret registry.domain.com'
+    end
+
+    after do
+      run 'kontena external-registry rm --force registry.domain.com'
+    end
+
+    it "returns a list" do
+      k = run('kontena external registry ls')
+      expect(k.code).to be_zero
+      expect(k.out.lines.first).to match(/NAME/)
+      expect(k.out.lines.last).to start_with 'registry.domain.com'
+      expect(k.out.lines.last).to include 'foo'
+      expect(k.out.lines.last).not_to include 'secret'
+      expect(k.out.lines.size).to eq 2
+    end
+
+    context '-q' do
+      it "outputs the registry name" do
+        k = run 'kontena external-registry ls -q'
+        expect(k.code).to be_zero
+        expect(k.out.strip).to eq 'registry.domain.com'
```

Comment:
TBH I don't know what's going on here, but I've seen this `stty` thing before... it must be something related to `TTY::Screen#size_from_stty` when run from Kommando... 

```
  3) external-registry ls when not empty -q outputs the registry name
     Failure/Error: expect(k.out.strip).to eq 'registry.domain.com'
     
       expected: "registry.domain.com"
            got: "stty: standard input\r\nregistry.domain.com"
     
       (compared using ==)
     
       Diff:
       @@ -1,2 +1,3 @@
       +stty: standard input
        registry.domain.com
       
     # ./spec/features/external_registry/list_spec.rb:44:in `block (4 levels) in <top (required)>'
```

Closest I can get to reproing this is:

```
/kontena/test # stty size < /dev/null > /dev/null 
stty: standard input: Not a tty
```

Reply:
Maybe updating the tty-screen gem would help?

Reply:
The `test/Dockerfile.test` image I'm using to run the e2e specs seems to have the latest version of `tty-screen`:

```
irb(main):008:0> TTY::Screen.method(:size_from_stty).source_location
=> ["/usr/lib/ruby/gems/2.3.0/gems/tty-screen-0.6.2/lib/tty/screen.rb", 207]
```

Reply:
The same syntax is used in other `-q` tests.

```ruby
#spec/features/stack/list_spec.rb:
  context 'quiet mode' do
    it "returns an installed stack name" do
      with_fixture_dir("stack/simple") do
        run 'kontena stack install --no-deploy'
      end
      k = run 'kontena stack ls -q'
      expect(k.code).to eq(0)
      expect(k.out.lines.size).to eq(1)
      expect(k.out.strip).to eq "simple"
    end
  end
```

Reply:
Yeah, looks like the same `stty: standard input` is now tripping up other e2e specs as well:

```
 [done] Creating stack simple      
stty: standard input
NAME       STACK               SERVICES   STATE         EXPOSED PORTS
⊝ simple   test/simple:0.1.0   1          initialized
 [done] Removing stack simple      
  returns an installed stack (FAILED - 1)
```

```
Failures:

  1) stack list returns an installed stack
     Failure/Error: expect(k.out.lines.size).to eq(2)
     
       expected: 2
            got: 3
     
       (compared using ==)
     # ./spec/features/stack/list_spec.rb:21:in `block (2 levels) in <top (required)>'
```